core/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -577,6 +577,16 @@ public void testAffixKeySetting() {
         assertFalse(listAffixSetting.match("foo"));
     }
 
+    public void testAffixAsMap() {
```

Comment:
Can you add a test for what happens if there are further subkeys? eg `foo.bar.baz.deep`